src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -546,13 +550,17 @@ private Task ProduceTagsAsync(TaggerContext<TTag> context, CancellationToken can
             TagSpanIntervalTree<TTag> latestTree,
             TagSpanIntervalTree<TTag> previousTree)
         {
-            var latestSpans = latestTree.GetSpans(snapshot);
-            var previousSpans = previousTree.GetSpans(snapshot);
+            using var _1 = ArrayBuilder<TagSpan<TTag>>.GetInstance(out var latestSpans);
+            using var _2 = ArrayBuilder<TagSpan<TTag>>.GetInstance(out var previousSpans);
```

Comment:
added pooling fo rthese as well.  this also means the iteration is allocfree.

Reply:
Why even mess with enumerators? We can just index directly into them now, right?

Reply:
my god.  you madman! will def change that :D

Reply:
actually, it didn't make it much better afaict.  i still need to track which builder and which index i'm working with.  And... that's just an enumerator.